base/sparse/sparsematrix.jl

```diff
@@ -926,7 +926,47 @@ end
 .<{Tv1,Ti1,Tv2,Ti2}(A_1::SparseMatrixCSC{Tv1,Ti1}, A_2::SparseMatrixCSC{Tv2,Ti2}) = broadcast!(<, spzeros( Bool, promote_type(Ti1, Ti2), broadcast_shape(A_1, A_2)...), A_1, A_2)
 .!={Tv1,Ti1,Tv2,Ti2}(A_1::SparseMatrixCSC{Tv1,Ti1}, A_2::SparseMatrixCSC{Tv2,Ti2}) = broadcast!(!=, spzeros( Bool, promote_type(Ti1, Ti2), broadcast_shape(A_1, A_2)...), A_1, A_2)
 
-
+## full equality
+function ==(A1::SparseMatrixCSC, A2::SparseMatrixCSC)
+    size(A1)!=size(A2) && return false
+    vals1 = nonzeros(A1)
+    vals2 = nonzeros(A2)
+    m, n = size(A1)
+    @inbounds for i = 1:n
+        nz1,nz2 = nzrange(A1,i), nzrange(A2,i)
+        if nz1==nz2
+            for j in nz1
+                vals1[j]!=vals2[j] && return false
+            end
+        else # As entries can be zeros this case needs checking too.
+            j1,j2 = first(nz1), first(nz2)
+            while j1<=last(nz1) && j2<=last(nz2)
+                r1,r2 = rowvals(A1)[j1], rowvals(A2)[j2]
```

Comment:
there's likely an advantage to hoisting out the `rowvals` into a separate variable outside of the loop